### DISCUSSION TITLE
The point of useRef

### DISCUSSION THREAD

**Discussion by honaseung** [honaseung/react-deep-dive#24] [Chapter3]
We talked about `useState` before. Do you remember? `useState` is used for remembering value. And if it's changed, Re-rendering is happened right? But what if you want to remember some values, but don't want to re-rendering when it's changed. `useRef` is for this. Remember the value, No re-rendering. Simple right?

Are you leaving? Hell no. There is one thing you should feel wierd. Why do you have to use this? We still can remember value without it. By declaraing it outside of component. What's different between `useRef` and declared value of outside component.

You can remember a value each of component when you use `useRef`. Because it's inside of component! So they have different value from each other! But when you declare a value from outside, components share that value. Because It's outside of component!

![image](https://github.com/honaseung/react-deep-dive/assets/53066347/53ce6935-53ad-41f4-9285-6c5f77f90960)

**Comment by honaseung** [comment]
useRef 는 useState 와 비슷하게 리액트에서 특정 값을 기억하기 위해 사용합니다. 그러나 동시에 리렌더링을 발생시키지 않기 위해 사용합니다. 이것은 component 외부에 값을 선언하는 것과는 다릅니다. 컴포넌트 외부의 선언은 컴포넌트끼리 공유가 되고 useRef 는 서로 공유가 되지 않기 때문입니다.

**Reply by chaminju927** [reply]
컴포넌트 외부에 선언된 값은 여러컴포넌트에서 자유롭게 사용이 가능하지만, useRef는 컴포넌트 내부에서 선언되기때문에 서로 공유가 안된다고 생각해도 무방한가요?

**Reply by honaseung** [reply]
네 정확합니다! 컴포넌트는 내부 스코프에서 생성되는 값이기 때문에 외부에서는 볼 수도 변경할 수도 없습니다! 이는 React 가 `closure` 의 개념을 이용하고 있기 때문입니다!
